src/MUST_analysis_tools/compute_homography_image_warp.py

```diff
@@ -0,0 +1,405 @@
+    allowed_corners = np.array([[208, 220], [770, 624]])
+    points_allowed = np.arange(len(object_points))[
+        (object_points[:, 0] > allowed_corners[0, 0]) &
+        (object_points[:, 0] < allowed_corners[1, 0]) &
+        (object_points[:, 1] > allowed_corners[0, 1]) &
+        (object_points[:, 1] < allowed_corners[1, 1])
+        ]
+    # points_allowed2 = (object_points[:, 0] > allowed_corners[0, 0]) & (object_points[:, 0] < allowed_corners[1, 0]) & (object_points[:, 1] > allowed_corners[0, 1])
+    # pt 50 (28 in accepted) -1 x pixel objects
+    # pt  (15 in accepted) [526, 259
+    # img_points = np.float32([img_points[i] for i in points_allowed])
+    # object_points = np.float32([object_points[i] for i in points_allowed])
+    # object_points_meters_allowed = np.float32([object_points_meters[i] for i in points_allowed])
+    object_points_3d = np.float32(np.append(object_points, np.zeros((len(object_points), 1)), 1))
+    object_points_norm = np.float32(np.append(object_points, np.ones((len(object_points), 1)), 1))
+    object_points_meters = np.float32((M_img_to_meters @ object_points_norm.T).T[:, 0:2])
+
+    # Camera matrix and distortion coefficients
+    # K = np.array([[1120, 0, 638], [0, 1330, 384], [0, 0, 1]])
+    # d = np.array([-0.562, 0.06, 0.01, 0.009, 0])
+    # Best for all points, 1.15 pixels
+    # K = np.array([[1140, 0, 639], [0, 1390, 388], [0, 0, 1]])
+    # d = np.array([-0.60, 0.1014, 0.0197, 0.0073, 0])
+    # BR, BL better
+    # K = np.array([[1140, 0, 550], [0, 1390, 300], [0, 0, 1]])
+    # d = np.array([-0.60, 0.1014, 0.0197, 0.0073, 0])
+    K = np.array([[1000, 0, 640], [0, 1000, -360], [0, 0, 1]])
+    # d = np.array([-0.50, 0.0, -0.001, 0.0011, 0.0])
+    d = np.array([0.0, 0.0, 0.0, 0.0, 0.0])
+
+    K = [[9.67481868e+02, 0.00000000e+00, 6.46144849e+02],
+         [0.00000000e+00, 1.02946621e+03, 3.49298579e+02],
+         [0.00000000e+00, 0.00000000e+00, 1.00000000e+00]]
+    d = [-0.80916231, 0.52110152, 0.0016839, -0.00374791, -0.14427669]
+    newcameramatrix = [[104.30975877, 0., 255.37791574],
+                       [0., 344.09854501, 244.28976651],
+                       [0., 0., 1.]]
+    H_und_nometers = [[-1.60736160e-01, 4.37323614e-02, 2.05116017e+01],
+         [2.16078130e-01, 3.09327514e-02, -6.94170640e+01],
+         [-7.65483284e-04, -5.58295093e-03, 1.00000000e+00]]
+
+    # min_reproj_error = 10000
+    # for i in range(20):
+    #     K_tmp, d_tmp, reproj_error = optimize_intrinsics(img_points, object_points, frame, reference)
+    #     if reproj_error < min_reproj_error:
+    #         K = K_tmp
+    #         d = d_tmp
+
+    # rep_error2, K2, d2, rvecs, tvecs = cv2.calibrateCamera([object_points_3d], [img_points], (1280, 720), None, None)
+    # newcameramatrix, roi = cv2.getOptimalNewCameraMatrix(K, d, (1280, 720), 1, (1280, 720))
+    # print(f'newmcaeramatrix: {newcameramatrix}')
+    # print(K)
+    # print(d)
+    # Undistort the camera image
```

Comment:
![alt text](image.png)

Reply:
I experience issue running this image_warp python script with below error:
```
Traceback (most recent call last):
  File "/home/dev/workspace/carma-analytics-fotda/src/MUST_analysis_tools/compute_homography_image_warp.py", line 405, in <module>
    main()
  File "/home/dev/workspace/carma-analytics-fotda/src/MUST_analysis_tools/compute_homography_image_warp.py", line 363, in main
    und = cv2.undistort(gray_camera_image, K, d, None, newcameramatrix)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
cv2.error: OpenCV(4.10.0) :-1: error: (-5:Bad argument) in function 'undistort'
> Overload resolution failed:
>  - cameraMatrix is not a numpy array, neither a scalar
>  - Expected Ptr<cv::UMat> for argument 'cameraMatrix'
```

Reply:
@dan-du-car try it now? I don't see that error